### ISSUE TITLE
Sanity tests not working on CPU machine

### ISSUE THREAD

**Issue by chauhang** [pytorch/serve#835]
After install from source, not able to run sanity tests on brand new ubuntu 18.04 cpu only machine (m3.large instance). 

Please see the error below: 

> ubuntu@ip-172-31-46-197:$ lsb_release -a
> No LSB modules are available.
> Distributor ID:	Ubuntu
> Description:	Ubuntu 18.04.5 LTS
> Release:	18.04
> Codename:	bionic
> ubuntu@ip-172-31-46-197:$ cat /proc/cpuinfo  | grep 'name'| uniq
> model name	: Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
> ubuntu@ip-172-31-46-197:~/serve$ python torchserve_sanity.py
> Traceback (most recent call last):
>   File "torchserve_sanity.py", line 4, in <module>
>     from ts_scripts.sanity_utils import test_sanity
>   File "/home/ubuntu/serve/ts_scripts/sanity_utils.py", line 3, in <module>
>     import nvgpu
> **ModuleNotFoundError: No module named 'nvgpu'**

**Comment by chauhang** [comment]
Also attempted to install on the machine using `python ts_scripts/install_dependencies.py --cuda=cu101` and then run the sanity tests again. Still the same error. Full logs below:

[cpu1804.log](https://github.com/pytorch/serve/files/5646373/cpu1804.log)

**Comment by lxning** [comment]
I followed the [instruction](https://github.com/pytorch/serve#install-torchserve-for-development), and then run "python torchserve_sanity.py" on CPU. There was no issue.

Could you please check if nvgpu is installed after "python ./ts_scripts/install_dependencies.py"?

**Comment by chauhang** [comment]
Problem was narrowed down to issue with python3 links changing. Following steps provided by @maaquib worked. Step#8 needs to be called again after install dependency on the fresh ubuntu 18.04 machines: 

1. sudo apt-get update && sudo apt-get install -y python3 python3-pip
2. git clone https://github.com/pytorch/serve
3. cd serve
4. sudo ln -sf /usr/bin/python3 /usr/bin/python
5. sudo ln -sf /usr/bin/pip3 /usr/bin/pip
6. pip install --upgrade pip
7. python ts_scripts/install_dependencies.py
8. **sudo ln -sf /usr/bin/python3 /usr/bin/python**
9. python ./ts_scripts/install_from_src.py
10. python ./torchserve_sanity.py
11. python test/regression_test.py

**Comment by jeffxtang** [comment]
I tested `python torchserve_sanity.py` on Windows 10 Pro with CPU, and got some other error (attached log). I installed the latest serve from source and can start the serve and get prediction from it using the commands [here](https://github.com/pytorch/serve#get-predictions-from-a-model).
[sanity_test_windows_log.txt](https://github.com/pytorch/serve/files/5646443/sanity_test_windows_log.txt)

**Comment by jeffxtang** [comment]
Complete test log is attached below..
[sanity_log_win_full.txt.zip](https://github.com/pytorch/serve/files/5646581/sanity_log_win_full.txt.zip)

**Comment by harshbafna** [comment]
@jeffxtang : You will need to run this as administrator on windows.

**Comment by jeffxtang** [comment]
@harshbafna 

From the Anaconda Powershell Prompt. where the sanity test script runs, I have this:
(base) PS C:\Users\Warrior\repos\serve> whoami
warrior\warrior

And from WIndows User Accounts, Warrior shows as Administator.

**Comment by dhaniram-kshirsagar** [comment]
@jeffxtang Can you please test windows native using [PR#836 Windows native](https://github.com/pytorch/serve/blob/build_fixes/docs/torchserve_on_win_native.md). NOTE that earlier, torchserve was never tested on windows 10. This PR has some fixes [scripts + doc] for windows 10 pro.

**Comment by jeffxtang** [comment]
@dhaniram-kshirsagar I grabbed PR 836 and did another test and still got the errors. Please see the attached log.
[sanity_log_win_full_pr836.txt](https://github.com/pytorch/serve/files/5649452/sanity_log_win_full_pr836.txt)

**Comment by harshbafna** [comment]
@jeffxtang: From the shared logs it looks like the TorchServe failed to create a worker on port number `9012` and ran into following error:

```
OSError: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
```

Could you please check if the port is in use by some other process on your system? In case you face more issues on windows please add the details on #828

**Comment by harshbafna** [comment]
Closing this as the issue related to install_dependency scripts has been fixed in #836 and merged to master. We will be tracking the windows related failures through #828 and #841

**Comment by jeffxtang** [comment]
> @jeffxtang: From the shared logs it looks like the TorchServe failed to create a worker on port number `9012` and ran into following error:
> 
> ```
> OSError: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
> ```
> 
> Could you please check if the port is in use by some other process on your system? In case you face more issues on windows please add the details on #828

I found that the permission error above was caused by my running the script actually not as Admin (forgot the right mouse click on Anaconda Powershell Prompt) even though I logged in as Admin. After I ran the sanity test script as Admin, the error above was gone but there's still 2 failed errors. I'll post the full log at #828.
